scripts/vpn_scan.sh

```diff
@@ -1,20 +1,25 @@
 #!/bin/sh
 
 # use with:
-# $ cat scripts/vpn_scan.sh | docker exec -i test_network_1_tailscale_1 ash
+# $ cat scripts/vpn_scan.sh | docker exec -i test_network_1-tailscale-1 ash
 
 apk add netcat-openbsd
 apk add lsof
 # lsof -i -P -n | grep LISTEN | grep -Eo ':(.+?)\s' | cut -c 2-
 
 # scan local ports
-tailscale status | grep "$(hostname)" | awk '{print $1}' | while read line; do
-    echo "Scanning $line"
-    nc -z -v "$line" 1-65535 2>&1 | grep succeeded
-done
+# uncomment to test against localhost
+# tailscale status | grep "$(hostname)" | awk '{print $1}' | while read line; do
+#     echo "Scanning $line"
+#     nc -z -v "$line" 1-65535 2>&1 | grep succeeded
+# done
 
-# scan other VPN IPs, this is slow so we are only including known ports for now
+# scan other VPN IPs, make sure to connect the containers to the other networks
+# otherwise they will route all the traffic through a relay and this will be really slow
+# for example:
+# docker network connect test_domain_1_default test_network_1-tailscale-1
 tailscale status | grep -v "$(hostname)" | awk '{print $1}' | while read line; do
     echo "Scanning $line"
-    nc -z -v "$line" 21 80 4000 8001 8011 8080 5050 5432 5555 5672 15672 41641 2>&1 | grep succeeded
+    nc -z -v "$line" 1-65535 2>&1 | grep succeeded
```

Comment:
Now we have a docker overlay network we can scan the entire port range fairly quickly so we don't need to manually input port numbers.